src/core/fetch.ts

```diff
@@ -0,0 +1,83 @@
+import queryStringify from '../utils/queryStringify';
+
+// eslint-disable-next-line no-shadow
+enum METHODS {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
+
+interface IOptions {
+  data?: any;
+  timeout?: number;
+  headers?: any;
+  method?: METHODS;
+}
+
+export class HTTPTransport {
+  private _BASE_URL = 'https://ya-praktikum.tech/api/v2';
+
+  get = (url: string, options: IOptions): Promise<XMLHttpRequest> => {
```

Comment:
Было бы здорово иметь возможно указывать тип для возвращаемого значения извне
```suggestion
 get = <T>(url: string, options: IOptions): Promise<T> => {
```